### DISCUSSION TITLE
LazyRunnable is now deprecated

### DISCUSSION THREAD

**Discussion by andreasdc** [netty/netty#13458] [Q&A]
How to use the new solution to not wakeup the thread when flushing? Previous code worked like that: https://github.com/PaperMC/Paper/blob/master/patches/server/0706-Optimise-non-flush-packet-sending.patch
https://github.com/netty/netty/commit/de876d197a945e2b3a4a96b90a65c00611b77806

**Comment by chrisvest** [comment]
You override the `SingleThreadedEventExecutor.wakesUpForTask` method to add the decision logic. The `Epoll/KQueue/NioEventLoop` classes are subclasses of the `SingleThreadedEventExecutor` class, so in practice you'll extend the event loop type you want to use.

**Comment by andreasdc** [comment]
I see the EventLoopGroup and EpollEventLoopGroup, should I create custom class for it and override it there?

**Reply by chrisvest** [reply]
Yes, if you use the `*EventLoopGroup` classes, you need to override the `newChild` method such that it produces an `EventLoop` that has the `wakesUpForTask` logic you want.

**Reply by andreasdc** [reply]
I can't figure out where should I head

**Comment by andreasdc** [comment]
Any hints? Because right now LazyRunnable doesn't work at all

**Comment by andreasdc** [comment]
Can anyone help? It's game breaking issue.

**Comment by chrisvest** [comment]
Cast the `eventLoop()` to an `AbstractEventExecutor` and use `lazyExecute()` instead of `execute()`.

**Reply by andreasdc** [reply]
Thank you so much! That seems to work, basically it doesn't wakeup the netty's thread, right?

**Comment by normanmaurer** [comment]
YesAm 03.07.2023 um 19:44 schrieb andreasdc ***@***.***>:﻿
Thank you so much! That seems to work, basically it doesn't wakeup the netty's thread, right?
